### PR TITLE
[WFLY-11959] AbstractSimpleApplicationClientTestCase pollutes the build/target/ installation

### PR DESCRIPTION
**Pull request by lukasburda**
JIRA: https://issues.jboss.org/browse/WFLY-11959
- Change distribution which basic tests used and polluted to one which is created especially for them

### CODE REVIEW

**Comment by kabir** [comment]
This is ok to test

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AbstractSimpleApplicationClientTestCase.java [hunk 1]
```diff
@@ -85,4 +87,22 @@ public void testAppClientJBossDescriptor() throws Exception {
             wrapper.quit();
         }
     }
+
+    @AfterClass
+    public static void afterClass() {
+        String appClientDir = System.getProperty("jboss.dist") + File.separator + "appclient";
+        deleteDirectory(new File(appClientDir + File.separator + "tmp"));
+        deleteDirectory(new File(appClientDir + File.separator + "data"));
+    }
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AbstractSimpleApplicationClientTestCase.java]: I may be misunderstanding the JIRA, but I don't think we want to do this. It's possible there is useful data in here if tests fail.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/util/AppClientWrapper.java [hunk 2]
```diff
@@ -201,6 +201,7 @@ private String getAppClientCommand() throws Exception {
                 " org.jboss.as.appclient" +
                 " -Djboss.server.base.dir="+ asDist + "/appclient" +
                 " -Djboss.home.dir="+ asDist +
+                " -Djboss.server.log.dir=" + System.getProperty("jboss.server.log.dir") +
```
Comment (jamezp) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/util/AppClientWrapper.java]: I don't think we need this property. However we would need the following two:
```
"-Dorg.jboss.boot.log.file=" + asDist + "/appclient/log/appclient.log" +
"-Dlogging.configuration=file:" + asDist + "/appclient/configuration/logging.properties" +
```

File: testsuite/integration/pom.xml [hunk 3]
```diff
@@ -354,6 +354,7 @@
                         <jbossas.ts.dir>${jbossas.ts.dir}</jbossas.ts.dir>
                         <jbossas.project.dir>${jbossas.project.dir}</jbossas.project.dir>
                         <jboss.dist>${jboss.dist}</jboss.dist>
+                        <jboss.server.log.dir>${basedir}/target/server-logs</jboss.server.log.dir>
```
Comment (jamezp) [inline testsuite/integration/pom.xml]: Given the above changes we would likely not need this.

**Review by jamezp — CHANGES_REQUESTED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AbstractSimpleApplicationClientTestCase.java (hunk 1 above)
Comment (bstansberry) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AbstractSimpleApplicationClientTestCase.java]: If what the PR is doing is cleaning up data in build/target/wildfly-xxx, that's not what I'm looking for. There should be nothing written there.

Ideally the test doesn't deal with that directory at all.

**Comment by lukasburda** [comment]
Thanks for the review. I will solve that recently.

**Comment by lukasburda** [comment]
retest this please

File: testsuite/pom.xml [hunk 4]
```diff
@@ -231,7 +231,6 @@
                                            <exclude>bin/*.bat</exclude>
                                            <exclude>bin/*.ps1</exclude>
                                            <exclude>docs/</exclude>
-                                           <exclude>modules/</exclude>
```
Comment (bstansberry) [inline testsuite/pom.xml]: We can't do this; copying the modules all over puts too much IO strain on the CI servers.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Perhaps the thing to do here is to leave jboss.dist as is and use the jboss.inst property in AppClientWrapper for paths that are meant to be under testsuite/integaration/basic/target/wildfly. Use jboss.inst for those that are meant to be under build/target/wildfly-xxx.

" -Djboss.modules.dir="+ asInst + "/modules" +


" -mp "+ asInst + "/modules" +

**Comment by darranl** [comment]
This PR seems to have stalled since June so I am going to close it for now, please feel free to re-open if you are ready to continue or submit a PR based on the comments so far.

**Pull request closed without merging**